### ISSUE TITLE
React dependency

### ISSUE THREAD

**Issue by eviltik** [protectwise/troika#134]
Hi !

The doc say "Note: Currently the only entry point provided is a React component called <Canvas3D>. This assumes your application uses React, and you're adding a Troika 3D canvas into it. In the future we plan to provide entry points for other frameworks, a web component, and a pure JS entry point function. Contributions are welcome."

Is there a particular reason that troika need React Canvas3D rather than basic threejs renderer.domElement ?

Technically, would it be possible to remove this dependency ?

Thank you

**Comment by lojjic** [comment]
> Technically, would it be possible to remove this dependency?

Yes, definitely -- The `Canvas3D` component is really just a convenience wrapper around a `World3DFacade` object. You could certainly create and use a `World3DFacade` yourself without the React bits, but the wrapper does some extra "glue" things:

- Creates the `canvas` element and surrounding DOM structure
- Rendering of HtmlOverlay contents outside the canvas, if present
- Error handling and fallback rendering for WebGL initialization errors

As I wrote in the docs I definitely want to create additional entry points that do those things, and/or abstract it differently to make the wrappers even lighter. I think my first would be a custom element (web component), but I don't have an ETA for getting to that myself.
